src/metpy/calc/tools.py

```diff
@@ -781,6 +781,41 @@ def take(indexer):
     return take
 
 
+@exporter.export
+def find_local_extrema(var, nsize, extrema):
+    r"""Find the local extreme (max/min) values of a 2D array.
+
+    Parameters
+    ----------
+    var : `xarray.DataArray`
+        The variable to locate the local extrema using the nearest method
+        from the maximum_filter or minimum_filter from the scipy.ndimage module.
+    nsize : int
```

Comment:
Really wish we could somehow choose sane defaults for this, but I have no idea how to begin to deal with that (unless we require xarray/coordinates...)

Reply:
Yeah, that is the difficult part because it all relies on the grid point spacing, which would be more feasible if we required xarray to be able to handle some things - might be something to back into the declarative implementation to help there.

Reply:
I think given the general move away from requiring xarray here, putting the nicety in declarative makes sense.